api/src/main/java/org/openmrs/module/patientflags/api/FlagService.java

```diff
@@ -139,6 +139,26 @@
 	@Transactional(readOnly = true)
 	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)
 	public Flag getFlag(Integer flagId);
+
+	/**
+	 * Retrieve a specific Flag based on its uuid
+	 *
+	 * @param uuid
+	 * @return the flag which matches the specific uuid
+	 */
+	@Transactional(readOnly = true)
+	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)
+	public Flag getFlagByUuid(String uuid);
+
+	/**
+	 * Retrieve a specific Flag based on its name
+	 *
+	 * @param name
+	 * @return the flag which matches the specific name
+	 */
+	@Transactional(readOnly = true)
+	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)

```

Comment:
What is "Test Flags" privilege for?

Reply:
@rkorytkowski its for previewing the patients who match the flags

Reply:
Hasn't our convention been "View" instead of "Test"? That is "View Flags".

Reply:
@dkayiwa based on my understand the Privilege should actually be "Preview Flags" rather than test which is what I think is meant to be